compiler/rustc_const_eval/src/interpret/step.rs

```diff
@@ -251,9 +252,14 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
 
             Len(place) => {
                 let src = self.eval_place(place)?;
```

Comment:
You should be able to do `src.place_to_op(src)?`, which will give you a `OpTy`, and then call `len` on that. That would avoid `force_allocation`.